src/reconcile/rec_visibility.c

```diff
@@ -654,6 +654,7 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *first_upd
          * first checkpoint with the precise mode.
          */
         if (F_ISSET(conn, WT_CONN_PRECISE_CHECKPOINT) &&
+          r->rec_start_pinned_stable_ts != WT_TS_NONE &&
```

Comment:
This is not correct in two cases:

* We first do a non-timestamped checkpoint then a timestamped checkpoint.
* We evict before we set the stable timestamp and then run a timestamped checkpoint.

We need to discuss what should be the behaviour of non-timestamped checkpoint when precise checkpoint is enabled or we should not allow non-timestamped checkpoint in this case.